app/shared/routes.jsx

```diff
@@ -42,6 +44,30 @@ let getRoutes = store => {
       })
   }
 
+  function getPageAndEmail (nextState, replace, callback) {
+    store.dispatch(notifyByEmail())
```

Comment:
Need to use the following pattern...

```
  let p1 = store.dispatch(...);
  let p2 = store.dispatch(...);

  Promise.all([p1, p2]).then(() => {
    callback();
  }).catch(err => {
    console.log(err);
    // error pushed to state
    callback();
  })
```

Reply:
This probably needs applying to both notification functions @justan0therdave : )